File: frontend/app/types/association/AssociationRequestResponseDto.ts
import type { RequestStatus } from "./RequestStatus";

export type AssociationRequestResponseDto = {
    "id": number,
    "patient": {
        "id": number,
        "name": string,
        "email": string,
        "userType": string
    },
    "relation": {
        "id": number,
        "name": string,
        "email": string,
        "userType": string
    },
    "type": string,
    "status": RequestStatus,
    "createdAt": string,
    "respondedAt": string,
    "creatorEmail": string,
    "responderEmail": string
};

File: backend/src/main/java/com/tcc/alzheimer/dto/Association/AssociationRequestResponseDto.java
package com.tcc.alzheimer.dto.Association;

import java.time.LocalDateTime;

import com.tcc.alzheimer.model.enums.RequestStatus;
import com.tcc.alzheimer.model.enums.RequestType;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor

public class AssociationRequestResponseDto {
    private Long id;
    private String patientEmail;
    private String relationEmail;
    private RequestType type;
    private RequestStatus status;
    private LocalDateTime createdAt;
    private LocalDateTime respondedAt;
    private String creatorEmail;
    private String responderEmail;
}
